src/xrpld/app/tx/detail/Transactor.cpp

```diff
@@ -1035,6 +1059,13 @@ Transactor::operator()()
                         (before->getType() == ltCREDENTIAL))
                         expiredCredentials.push_back(index);
                 }
+
+                if (doWasmData && before && after &&
```

Comment:
You must not modify an object if error returned.

Reply:
This is outlining an exception being made for WASM code, where the `Data` field is still updated.